File: _739_daily_temperatures/main.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> result(temperatures.size());
        result[temperatures.size() - 1] = 0;

        for (int i = temperatures.size() - 2; i >= 0; i--) {
            result[i] = 1;
            int j = i + 1;

            while (temperatures[j] <= temperatures[i]) {
                if (result[j] == 0) {
                    result[i] = 0;
                    break;
                }

                result[i] += result[j];
                j += result[j];
            }
        }

        return result;
    }
};

File: _739_daily_temperatures/main.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        results = [0 for _ in range(len(temperatures))]
        for i in range(len(temperatures) - 2, -1, -1):
            results[i] = 1
            j = i + 1

            while (temperatures[j] <= temperatures[i]):
                if results[j] == 0:
                    results[i] = 0
                    break

                results[i] += results[j]
                j += results[j]

        return results